vmm/src/cpu.rs

```diff
@@ -680,14 +680,23 @@ impl CpuManager {
             .sgx_epc_region()
             .as_ref()
             .map(|sgx_epc_region| sgx_epc_region.epc_sections().values().cloned().collect());
+
+        let topology = self.config.topology.clone().map_or_else(
+            || {
+                #[cfg(feature = "mshv")]
+                if matches!(hypervisor.hypervisor_type(), HypervisorType::Mshv) {
+                    return Some((1, self.boot_vcpus(), 1));
```

Comment:
Why the special case here? Shouldn't this be pushed into MSHV code?

Reply:
This part serves the default behavior case where no topology argument has been passed to Cloud Hypervisor. In that case, this mimics the KVM behavior when the hypervisor in use is MSHV. I guess, moving this into the `mshv` crate is possible but would require more effort and some more clarity. Some closer points on this:

 a. The required CPU info in the `mshv` crate seems currently missing (boot vcpus, etc.)
 b. It is needed to know, if some topology has been passed or not
 c. More conditions might be involved, not always Cloud Hypervisor related, at least with regard to what the current partition used (exo partition, root nested, root baremetal)
 b. The default handling in `mshv` crate might not have to be the one to mimic KVM

What is currently in `mshv` crate is only the logic that would iterate through the passed CPUID entries and execute the `HvCallRegisterInterceptResult`. The place where this special condition is put in Cloud Hypervisor is it is known the topology has not been passed so it'll set the behavior same as KVM. It seems to me, that there would be no 100% clarity yet about how to correctly handle all the possible conditions in the `mshv` crate, so i'd suggest the simplest is to have this special case.

Thanks

Reply:
@liuw would you have time to continue on this one please?

Thanks